### PR TITLE
Api docs generate as md

### PR DESCRIPTION
**Pull request by queil**
This PR addresses #393 and extends it by providing ApiDocs generation in Markdown (to use with a static site generator). There are still quite a few things that are not right (and no tests but [this](https://queil.github.io/yzl/reference/yzl-core-yzl/)) but I hope we can resolve those.

### CODE REVIEW

File: src/FSharp.Formatting.ApiDocs/GenerateMarkdown.fs [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+        yield! renderMembers "Active patterns" "Active pattern" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.ActivePattern))
+        yield! renderMembers "Union cases" "Union case" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.UnionCase))
+        yield! renderMembers "Record fields" "Record Field" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.RecordField))
+        yield! renderMembers "Static parameters" "Static parameters" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.StaticParameter))
+        yield! renderMembers "Constructors" "Constructor" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.Constructor))
+        yield! renderMembers "Instance members" "Instance member" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.InstanceMember))
+        yield! renderMembers "Static members" "Static member" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.StaticMember))
+    ]
+
+  let categoriseEntities (nsIndex: int, ns: ApiDocNamespace, suppress) =
```
Comment (dsyme) [inline src/FSharp.Formatting.ApiDocs/GenerateMarkdown.fs:303]: I think this code can be shared between Html and Markdown genreators?

Comment (queil) [reply src/FSharp.Formatting.ApiDocs/GenerateMarkdown.fs:303]: Indeed, it needs a bit of refactoring.

File: src/FSharp.Formatting.ApiDocs/GenerateDoc.fs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+module internal FSharp.Formatting.ApiDocs.GenerateDoc
```
Comment (queil) [inline src/FSharp.Formatting.ApiDocs/GenerateDoc.fs:1]: This probably could end up in `GenerateModel.fs`

**Comment by cartermp** [comment]
@queil are you still interested in taking this forward?

**Comment by queil** [comment]
@cartermp Sure, I should have some more time over holidays to finish it. Any suggestions are welcome too.

**Comment by cartermp** [comment]
No worries! Don't feel obligated to spend holiday time on this unless you're bored :)

File: src/FSharp.Formatting.ApiDocs/GenerateDoc.fs (hunk 2 above)
Comment (queil) [reply src/FSharp.Formatting.ApiDocs/GenerateDoc.fs]: Renamed the module to `Categorise`.

**Comment by queil** [comment]
Ok, I think this is as far as I can get this time without refactoring `GenerateModel.fs` which is coupled with html (as far as rendering xml docs is concerned). The in-line HTML renders mostly fine so that should not be a problem.

**Review by cartermp — APPROVED** [review]
I think the changes here are good! I will want @dsyme to take a look, but he is offline until next week. I'll keep this in my notifications until that time

File: src/FSharp.Formatting.ApiDocs/GenerateMarkdown.fs (hunk 1 above)
Comment (dsyme) [reply src/FSharp.Formatting.ApiDocs/GenerateMarkdown.fs]: Good work on resolving this, thank you

**Comment by dsyme** [comment]
@queil I think this looks really good

Can you add some tests as well please?  Even just a couple of smoke tests

**Comment by queil** [comment]
@dsyme Sure.

**Comment by queil** [comment]
@dsyme Working on the tests I think I have found a bug in `UsageHtml` generation for `op_Implicit` - [example](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-core-fsharpfunc-2.html#op_Implicit) - shouldn't there be a space in between `op_Implicit` and the parameter?

**Comment by queil** [comment]
@dsyme I've just added tests but found a few more issues to solve.

**Comment by queil** [comment]
Oh, I think the build is failing because I've used string interpolation.

**Comment by dsyme** [comment]
@queil Ready?

**Comment by queil** [comment]
@dsyme Yes.

**Comment by dsyme** [comment]
Thanks!

**Pull request merged**